sklearn/manifold/_spectral_embedding.py

```diff
@@ -86,6 +88,8 @@ def _graph_is_connected(graph):
         True means the graph is fully connected and False means not.
     """
     if sparse.issparse(graph):
+        # `connected_components` only works with 32-bits indices
+        graph = check_array(graph, accept_sparse=True, accept_large_sparse=False)
```

Comment:
This has been addressed recently by https://github.com/scipy/scipy/pull/18913 which has been included in https://github.com/scipy/scipy/pull/19279.
```suggestion
        # Before SciPy 1.11.3, `connected_components` only supports 32-bits indices.
        # PR: https://github.com/scipy/scipy/pull/18913
        # First integration in 1.11.3: https://github.com/scipy/scipy/pull/19279
        # TODO(jjerphan): Once SciPy 1.11.3 is the minimum supported version, use
        # `accept_large_sparse=True`.
        accept_large_sparse = sp_version >= parse_version("1.11.3")
        graph = check_array(
                    graph,
                    accept_sparse=True,
                    accept_large_sparse=accept_large_sparse,
                )
```

This might need formatting.